### DISCUSSION TITLE
Can extension boot files attach functions to the app instance lifecycle hooks

### DISCUSSION THREAD

**Discussion by julie777** [quasarframework/quasar#10469] [Quasar App Extensions]
I understand the use of lifecycle methods in Vue components. 
I don't understand how the lifecycle hooks for the app instance interact with the boot files for extensions.

Given that I have create a boot file, using Amplify as an example, then 
- code outside of the exported boot function is run when the file is loaded. I put `Amplify.configure` here. I also put `DataStore.start` here as it doesn't depend on any Quasar/Vue setup being completed.
- code inside the `export default boot` function is executed after the router default export is run and has access to the app instance, etc. so it can call app.use() if needed.
So essentially the boot file code runs as part of the app created lifecycle hook.

**What I want to know is how to run some boot file code when the app is being destroyed.** 

Just imagine that I need to release some resource. This could be to disconnect from bluetooth. It could be to clear all data for the current user from the cache or local storage. There are numerous things that might need to be done during app shutdown. (I am new to Quasar/Vue so my perception of reality is based on writing typical desktop applications, where asking a user if they want to save a file before exiting an app is commonplace.)

**Comment by MatthewSH** [comment]
You'd need to handle the native `beforeunload` in the window too by the way since the page itself is destroyed and none of those hooks are run and then you can in you're main `App.vue` you could use the `unmounted` lifecycle hook.
As for doing it from a boot file? Not sure if you can because you'd have to grab the app's root component and then add it onto that, and at that rate, you might as well do it within your `App.vue` unless you really have some reservation about that. I could  be wrong though.

**Comment by Mint-Joy** [comment]
If you are a desktop application specialist,
then imagine that you require an OnResetButton hook.
You will not have time to disconnect anything, if there is any sensitive data, store it in a destructible place, for example, in a state store. If you need to save some data, use localstorage, etc. Design your application like you can hit reset at any second and the reset doesn't only affect the storage.
